src/app/components/Answers.js

```diff
@@ -1,27 +1,57 @@
 import isAnswerCorrect from '../logic/isAnswerCorrect';
+import elementFactory from '../utils/elementFactory';
+import validateString from '../utils/validateString';
 
-const Answers = (answers, correctAnswer, callback) => {
-  const answersElement = document.createElement('div');
-  answersElement.classList.add('answers');
-  answers.forEach((answer) => {
-    const answerElement = document.createElement('div');
-    answerElement.classList.add('button');
-    answerElement.classList.add('button__answer');
-    answerElement.addEventListener('click', () => {
-      if (answersElement.hasAttribute('disabled')) {
-        return;
-      }
-      const correct = isAnswerCorrect(answer, correctAnswer);
-      if (correct) {
-        answerElement.classList.add('button__answer--success');
-      } else {
-        answerElement.classList.add('button__answer--fail');
-      }
-      answersElement.setAttribute('disabled', true);
-      callback(answer, correct);
-    });
-    answerElement.innerText = answer;
-    answersElement.append(answerElement);
+const createAnswerElement = (
+  answerText,
+  correct,
+  answersElement,
+  onAnsweredQuestion,
+  callback,
+) => {
+  const answerElement = elementFactory('div', {
+    className: 'button button__answer',
+  });
+  answerElement.addEventListener('click', () => {
+    if (answersElement.hasAttribute('disabled')) {
+      return;
+    }
+    answerElement.classList.add(
+      `button__answer--${correct ? 'success' : 'fail'}`,
+    );
+
+    answersElement.setAttribute('disabled', true);
+    setTimeout(
+      () => onAnsweredQuestion(answerText, correct, callback),
```

Comment:
I'm not sure about this callback, let me know @Enessetere if that's what you or/and @DominikNowak need.